include/deal.II/base/bounding_box.h

```diff
@@ -77,16 +77,35 @@ class BoundingBox
   const std::pair<Point<spacedim,Number>,Point<spacedim,Number>> &get_boundary_points () const;
 
   /**
+   * Check if the current object and @p other_bbox are neighbours i.e. if the boxes
+   * have dimension spacedim, check if their insersection is non empty and has at least dimension
+   * spacedim-2 (i.e. a point in dimension 2, an edge in dimension 3)
+   *
+   * Return values:
+   * - 0 : not neighbours
+   * Values which are non zero are used for neighbours, in particular:
+   * - 1 : neighbours
+   * - 2 : neighbours which can be expressed with a single Bounding Box, e.g.
+   *  @code
+   *  .--V--W    .-----V
+   *  |  |  | =  |     |
+   *  V--W--.    V-----.
+   *  @endcode
+   */
+  unsigned int is_neighbour(const BoundingBox<spacedim,Number> &other_bbox, const double &err=1e-12) const;
```

Comment:
why do you need `err` as a parameter? You could as well used internally
```
100. * std::numeric_limits<double>::epsilon
```
or alike.